storage/innobase/include/gis0type.h

```diff
@@ -66,10 +66,7 @@ typedef std::vector<rtr_rec_t, ut_allocator<rtr_rec_t> >	rtr_rec_vector;
 
 /* Structure for matched records on the leaf page */
 typedef	struct matched_rec {
-	byte*		bufp;		/*!< aligned buffer point */
-	byte		rec_buf[UNIV_PAGE_SIZE_MAX * 2];
```

Comment:
This is a suggestion. Perhaps we can get rid of rtr_info->heap. It is used for one time allocation of matched_rec  of size less than 1k.